Source/santad/EventProviders/SNTEndpointSecurityManagerTest.mm

```diff
@@ -0,0 +1,126 @@
+/// Copyright 2021 Google Inc. All rights reserved.
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///    http://www.apache.org/licenses/LICENSE-2.0
+///
+///    Unless required by applicable law or agreed to in writing, software
+///    distributed under the License is distributed on an "AS IS" BASIS,
+///    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+///    See the License for the specific language governing permissions and
+///    limitations under the License.
+#import <XCTest/XCTest.h>
+#import <bsm/libbsm.h>
+
+#import "Source/common/SNTConfigurator.h"
+#import "Source/santad/EventProviders/SNTEndpointSecurityManager.h"
+
+// Must be imported last to overload libEndpointSecurity functions.
+#import "Source/santad/EventProviders/EndpointSecurityTestUtil.h"
+
+const NSString *const kEventsDBPath = @"/private/var/db/santa/events.db";
+const NSString *const kRulesDBPath = @"/private/var/db/santa/rules.db";
+
+@interface SNTEndpointSecurityManagerTest : XCTestCase
+@property SNTEndpointSecurityManager *snt;
+@end
+
+@implementation SNTEndpointSecurityManagerTest
+
+- (void)setUp {
+  [super setUp];
+  fclose(stdout);
+}
+
+- (void)testDeleteRulesDB {
+  __block int wantNumResp = 2;
+
+  MockEndpointSecurity *mockES = [MockEndpointSecurity mockEndpointSecurity];
+  [mockES reset];
+  SNTEndpointSecurityManager *snt = [[SNTEndpointSecurityManager alloc] init];
+
+  [snt setLogCallback:^(santa_message_t m) {
+    return;
+  }];
+  [snt setDecisionCallback:^(santa_message_t m) {
+    return;
+  }];
+
+  XCTestExpectation *expectation =
+    [self expectationWithDescription:@"Wait for santa's Auth dispatch queue"];
+
+  __block NSMutableArray<ESResponse *> *events = [NSMutableArray array];
+  [mockES registerResponseCallback:^(ESResponse *r) {
+    @synchronized(self) {
+      [events addObject:r];
```

Comment:
Instead of storing the responses and counting them, could you verify that a 'deny' decision arrived with the correct path?